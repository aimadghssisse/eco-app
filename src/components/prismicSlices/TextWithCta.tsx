/**
 * Prismic Slice - Landing page Hero Component
 */

import * as React from "react"
import {Link} from "gatsby";

const TextWithCta = ({ data }) => {
    return (
        <div className="text-center">
            <h1 className="text-2xl mt-12 mb-2 font-extrabold">{data.primary.title}</h1>
            <p className="mb-4">{data.primary.subtitle}</p>
            <div className="mb-4">
                <Link to="/property-type">
                    <button className="text-center px-4 py-2 text-xl font-extrabold text-white leading-normal bg-green-light rounded-lg hover:bg-green-dark">{data.primary.button_1_text}</button>
                </Link>
            </div>
            <a href={data.primary.button_2_link.url} className="text-green hover:underline">{data.primary.button_2_text}</a>
        </div>
    )
}

export default TextWithCta