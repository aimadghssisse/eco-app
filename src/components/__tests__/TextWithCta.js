import React from "react"
import renderer from "react-test-renderer"

import TextWithCta from "../prismicSlices/TextWithCta"
import {Link} from "gatsby";

describe("TextWithCta", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <div className="text-center mt-6">
                    <h1 className="text-2xl mt-12 mb-2 font-extrabold">data.primary.title</h1>
                    <p className="mb-4">data.primary.subtitle</p>
                    <div className="mb-4">
                        <Link to="/property-type">
                            <button type="link" className="btn-lg">data.primary.button_1_text</button>
                        </Link>
                    </div>
                    <a href="data.primary.button_2_link.url">data.primary.button_2_text</a>
                </div>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})