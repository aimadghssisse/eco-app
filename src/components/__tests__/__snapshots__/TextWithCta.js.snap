// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithCta renders correctly 1`] = `
<div
  className="text-center mt-6"
>
  <h1
    className="text-2xl mt-12 mb-2 font-extrabold"
  >
    data.primary.title
  </h1>
  <p
    className="mb-4"
  >
    data.primary.subtitle
  </p>
  <div
    className="mb-4"
  >
    <a
      href="/property-type"
    >
      <button
        className="btn-lg"
        type="link"
      >
        data.primary.button_1_text
      </button>
    </a>
  </div>
  <a
    href="data.primary.button_2_link.url"
  >
    data.primary.button_2_text
  </a>
</div>
`;

exports[`TextWithCta renders correctly 1`] = `
<div
  className="text-center mt-6"
>
  <h1
    className="text-2xl mt-12 mb-2 font-extrabold"
  >
    data.primary.title
  </h1>
  <p
    className="mb-4"
  >
    data.primary.subtitle
  </p>
  <div
    className="mb-4"
  >
    <a
      href="/property-type"
    >
      <button
        className="btn-lg"
        type="link"
      >
        data.primary.button_1_text
      </button>
    </a>
  </div>
  <a
    href="data.primary.button_2_link.url"
  >
    data.primary.button_2_text
  </a>
</div>
`;
